doc/api_ref/bigint.rst

```diff
@@ -199,18 +200,15 @@ declarations in ``botan/bigint.h`` and ``botan/numthry.h``.
 
       Return absolute value of ``*this``
 
-   .. cpp:function:: void binary_encode(uint8_t buf[]) const
+   .. cpp:function:: void serialize_to(std::span<uint8_t> buf)
 
       Encode this BigInt as a big-endian integer. The sign is ignored.
 
-   .. cpp:function:: void binary_encode(uint8_t buf[], size_t len) const
+      There must be sufficient space to encode the entire integer in ``buf``.
+      If ``buf`` is larger than required, sufficient zero bytes will be
+      prefixed.
 
-      Encode this BigInt as a big-endian integer. The sign is ignored.
-      If ``len`` is less than ``bytes()`` then only the low ``len``
-      bytes are output. If ``len`` is greater than ``bytes()`` then
-      the output is padded with leading zeros.
-
-   .. cpp:function:: void binary_decode(uint8_t buf[])
+   .. cpp:function:: void assign_from_bytes(std::span<const uint8_t> buf)
```

Comment:
Yeah I see what you mean. Certainly simple assignment is more idiomatic. But being able to set a new value without an extra allocation is sometimes rather critical - but mostly for us, rather than for our users. I made it private, with a public `_` prefixed forwarder for the callers outside of `BigInt` itself. We can always make it public later if it seems useful...